src/ivc/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -857,114 +857,151 @@ mod tests {
             .unwrap();
     }
 
+    const LENGHT: usize = 5;
+    const TEST_FOLD_ROUND: usize = 3;
+    const K: u32 = 18;
+
     fn get_table_data() -> (TableData<Base>, MainGateConfig<T>) {
-        let mut td = TableData::new(17, vec![]);
+        let mut td = TableData::new(K, vec![]);
         let _ = td.cs.instance_column();
         let config = td.prepare_assembly(MainGate::<Base, T>::configure);
 
         (td, config)
     }
 
-    #[test_log::test]
-    fn fold_W_test() {
-        const LENGHT: usize = 5;
+    fn generate_random_input(mut rnd: impl Rng) -> Vec<C1> {
+        iter::repeat_with(|| C1::random(&mut rnd))
+            .take(LENGHT)
+            .collect::<Vec<_>>()
+    }
 
-        let (mut td, config) = get_table_data();
+    fn assign_curve_points<C, const T: usize>(
+        ctx: &mut RegionCtx<C::Base>,
+        ecc: &EccChip<C, C::Base, T>,
+        points: &[C],
+        var_prefix: &str,
+    ) -> Result<Vec<AssignedPoint<C>>, halo2_proofs::plonk::Error>
+    where
+        C: CurveAffine,
+        C::Base: PrimeFieldBits + FromUniformBytes<64>,
+    {
+        points
+            .iter()
+            .enumerate()
+            .map(|(i, point)| ecc.assign_from_curve(ctx, || format!("{var_prefix}[{i}]"), point))
+            .collect()
+    }
 
-        let mut rnd = rand::thread_rng();
-        let input_W = iter::repeat_with(|| C1::random(&mut rnd))
-            .take(LENGHT)
-            .collect::<Vec<_>>();
+    struct Fixture {
```

Comment:
add comment to explain `Fixture`

Reply:
Where is it used?

Reply:
`Fixture` is the standard name for a test context that includes all the required fields. I will add a generic description at the top, however, this is a common term

It will be used in all tests

Reply:
Okay, we can just leave as it is.